stdlib/_thread.pyi

```diff
@@ -21,16 +22,29 @@ class LockType:
         self, type: type[BaseException] | None, value: BaseException | None, traceback: TracebackType | None
     ) -> None: ...
 
+@final
+class _ThreadHandle:
+    ident: int
+    def __init__(self, *args: Any, **kwargs: Any) -> None: ...
+    def join(self, timeout: float | None = -1) -> None: ...
+    def _set_done(self) -> None: ...
+    def is_done(self) -> bool: ...
+
 @overload
 def start_new_thread(function: Callable[[Unpack[_Ts]], object], args: tuple[Unpack[_Ts]], /) -> int: ...
 @overload
 def start_new_thread(function: Callable[..., object], args: tuple[Any, ...], kwargs: dict[str, Any], /) -> int: ...
-def interrupt_main() -> None: ...
+def interrupt_main(signum: Signals = Signals.SIGINT, /) -> None: ...  # noqa: Y011
 def exit() -> NoReturn: ...
 def allocate_lock() -> LockType: ...
 def get_ident() -> int: ...
 def stack_size(size: int = 0, /) -> int: ...
 
+if sys.version_info >= (3, 13):
+    def start_joinable_thread(
```

Comment:
This one is for internal use only, so I'm not sure whether it should be exposed in typings. The current docs says:

```text
*For internal use only*: start a new thread.

Like start_new_thread(), this starts a new thread calling the given function.
Unlike start_new_thread(), this returns a handle object with methods to join
or detach the given thread.
This function is not for third-party code, please use the
`threading` module instead. During finalization the runtime will not wait for
the thread to exit if daemon is True. If handle is provided it must be a
newly created thread._ThreadHandle instance.
```

Reply:
The entire module is only for internal use, though.

Reply:
Yes, but this one seems to be even more internal. Some I'm not really sure we want to add something that *looks* public in the internal module but should really not be used at all (especially with this kind of comment...). For instance, we didn't add `_make_thread_handle` because it was correctly prefixed by `_` (I think `start_new_thread` doesn't have an underscore because of historical reasons).